SPEC.md

```diff
@@ -98,61 +98,113 @@ A build tool is something which reads one or more input artifacts and writes one
 ## Specifications
 ### Artifact ID
 
-GitBOM uses the git object id (as a blob) for an artifact as its artifact ID.
+A hash function over the []byte of an artifact satisfy the canonical, unique, immutable requirements for an artifact identifier.
+
+The vast majority of the source code artifacts are already indexed by git object id (gitoids) as git objects of type blob.
+
+For this reason GitBOM uses the git object id (gitoid) (as a blob) for an artifact as its artifact ID.  
+
+Git currently supports two varieties of gitoids.  One is based on SHA1 and is in common use.  The other is based on SHA256 and has been very slow to garner adoption.  In order to distinguish between them we will use the gitoid uri prefixes of gitoid:blob:sha1 and gitoid:blob:sha256.
+
+All subsequent references to mandatory all identifiier types in this document should be interpretted to mean the list:
+
+- gitoid:blob:sha1
+- gitoid:blob:sha256
 
 ### GitBOM Document
 
-Each artifact has a GitBOM document that describes its immediate children consiting of a set of new line delimited records, one for each child, in lexical order.
+A GitBOM document describes the immediate children of an artifact in the Artifact Dependency Graph (ADG).
+
+A GitBOM document utilizes precisely one identifier type.
+
+### GitBOM identifier
+
+A GitBOM document is identified by computing its identifier as an artifact with the identifier type used for identifiers within the GitBOM document itself.
```

Comment:
I believe this is the result of this week's discussion. I may have misinterpreted something while seeking to provide further clarity.
```suggestion
A GitBOM Document is an artifact which can be identified by it's gitoid. The GitBOM Document Identifier must be generated using the same Gitoid Identifier Type as used within the GitBOM document itself.
```